### ISSUE TITLE
error on lzma uncompress r47026 with F1

### ISSUE THREAD

**Issue by Mafketel** [ScApi/P2812HNUFx-Pre-Build#5]
I had the same issue as this person
https://forum.openwrt.org/viewtopic.php?pid=293574#p293574

I've been trying to install OpenWrt onto my P-2812HNU-F1 following the instructions in the Wiki. Replacing the factory U-Boot went fine, but I'm running into troubles when trying to boot OpenWrt. It fails with the message LZMA: uncompress or overwrite error 7, before rebooting immediately after.
trying r45578 now

**Comment by ScApi** [comment]
I don't have F1 any more so I cannot test it but You can try and report any issues.
https://1drv.ms/f/s!Aqp_c2-AFc_t5VTBRH919QS1noUO

**Comment by Mafketel** [comment]
So uploaded the asc file and then tftpd the img you send me.
There is still an error but it is not 19 anymore it is now 1
The r45578 version worked btw

log r47026
```
ROM VER: 1.0.5
CFG 06
NAND
NAND Read OK

U-Boot SPL 2013.10-openwrt5 (Apr 30 2018 - 06:32:23)
SPL: initializing NAND flash
SPL: checking U-Boot image
SPL: loading U-Boot to RAM
SPL: decompressing U-Boot with LZO
SPL: jumping to U-Boot


U-Boot 2013.10-openwrt5 (Apr 30 2018 - 06:32:23) P-2812HNU-Fx

Board: ZyXEL P-2812HNU-Fx
SoC:   Lantiq VRX288 v1.1
CPU:   500 MHz
IO:    250 MHz
BUS:   250 MHz
BOOT:  NAND
DRAM:  128 MiB
NAND:  128 MiB
*** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Net:   ltq-eth
Hit any key to stop autoboot:  0
Wrong Image Format for bootm command
ERROR: can't get kernel image!
P-2812HNU-Fx # tftp 0x80800000 openwrt-lantiq-xrx200-P2812HNUF1-uImage
ltq_phy: addr 0, link 1, speed 1000, duplex 1
ltq_phy: addr 1, link 0, speed 10, duplex 0
ltq_phy: addr 17, link 0, speed 10, duplex 0
ltq_phy: addr 19, link 0, speed 10, duplex 0
ltq_phy: addr 5, link 0, speed 10, duplex 0
Using ltq-eth device
TFTP from server 192.168.1.2; our IP address is 192.168.1.1
Filename 'openwrt-lantiq-xrx200-P2812HNUF1-uImage'.
Load address: 0x80800000
Loading: #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         ############
done
Bytes transferred = 1724531 (1a5073 hex)
P-2812HNU-Fx # nand erase 0x60000 0x200000

NAND erase: device 0 offset 0x60000, size 0x200000
Erasing at 0x240000 -- 100% complete.
OK
P-2812HNU-Fx # nand write 0x80800000 0x60000 0x200000

NAND write: device 0 offset 0x60000, size 0x200000
 2097152 bytes written: OK
P-2812HNU-Fx # tftpboot 0x80800000 openwrt-lantiq-xrx200-P2812HNUF1-uImage-initramfs
ltq_phy: addr 0, link 1, speed 1000, duplex 1
ltq_phy: addr 1, link 0, speed 10, duplex 0
ltq_phy: addr 17, link 0, speed 10, duplex 0
ltq_phy: addr 19, link 0, speed 10, duplex 0
ltq_phy: addr 5, link 0, speed 10, duplex 0
Using ltq-eth device
TFTP from server 192.168.1.2; our IP address is 192.168.1.1
Filename 'openwrt-lantiq-xrx200-P2812HNUF1-uImage-initramfs'.
Load address: 0x80800000
Loading: T #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         ####################################
         545.9 KiB/s
done
Bytes transferred = 4508385 (44cae1 hex)
P-2812HNU-Fx # bootm 0x80800000
## Booting kernel from Legacy Image at 80800000 ...
   Image Name:   MIPS OpenWrt Linux-3.18.21
   Created:      2015-09-23  21:03:52 UTC
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    4508321 Bytes = 4.3 MiB
   Load Address: 80002000
   Entry Point:  80002000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... LZMA: uncompress or overwrite error 1 - must RESET b
ROM VER: 1.0.5
CFG 06
NAND
NAND Read OK

U-Boot SPL 2013.10-openwrt5 (Apr 30 2018 - 06:32:23)
SPL: initializing NAND flash
SPL: checking U-Boot image
SPL: loading U-Boot to RAM
SPL: decompressing U-Boot with LZO
SPL: jumping to U-Boot


U-Boot 2013.10-openwrt5 (Apr 30 2018 - 06:32:23) P-2812HNU-Fx

Board: ZyXEL P-2812HNU-Fx
SoC:   Lantiq VRX288 v1.1
CPU:   500 MHz
IO:    250 MHz
BUS:   250 MHz
BOOT:  NAND
DRAM:  128 MiB
NAND:  128 MiB
*** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Net:   ltq-eth
Hit any key to stop autoboot:  0
Wrong Image Format for bootm command
ERROR: can't get kernel image!
P-2812HNU-Fx #
```


  
log r45578

```
P-2812HNU-Fx # bootm 0x80800000
## Booting kernel from Legacy Image at 80800000 ...
   Image Name:   MIPS OpenWrt Linux-3.18.11
   Created:      2015-04-25   1:54:58 UTC
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    4707169 Bytes = 4.5 MiB
   Load Address: 80002000
   Entry Point:  80002000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK

Starting kernel ...

[    0.000000] Linux version 3.18.11 (sylwek@WRT) (gcc version 4.8.3 (OpenWrt/Linaro GCC 4.8-2014.04 r44945) ) #2 Sat Apr 25 03:54:35 CEST 2015
```

**Comment by Mafketel** [comment]
ps also tested r46993 now and also boots, did not test functionality yet.... I had some problems with the newest stable LEDE 17.01.4 P2812HNUF1-squashfs-sysupgrade.bin but that is not something for this issue.... for it is not related ;)
```

## Booting kernel from Legacy Image at 80800000 ...
   Image Name:   MIPS OpenWrt Linux-3.18.21
   Created:      2015-09-20   6:54:35 UTC
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    4442167 Bytes = 4.2 MiB
   Load Address: 80002000
   Entry Point:  80002000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK

Starting kernel ...

[    0.000000] Linux version 3.18.21 (sylwek@WRT) (gcc version 4.8.3 (OpenWrt/Linaro GCC 4.8-2014.04 r46993) ) #6 Sun Sep 20 08:53:55 CEST 2015
[    0.000000] SoC: VR9 rev 1.1
```

**Comment by ScApi** [comment]
r47026 probably had some error while compiling(someone reported broken kernel image), as they are bit old maybe try Lede/OpenWrt images ?

**Comment by Mafketel** [comment]
I did try the upgrade to the latest 17.0.1.4 directly but had some weird issue with the mac addresses not being able to be set and no traffic on the wan port.
I tried again and it is sort of working now and am now in a testing fase.
My main concern was the fact that, on the wiki page your directed to the scapi installation.
And the most recent one is not working with that error, together with the very long wait because of tera term, realterm goes a lot faster 30 minutes vs multiple hours. After sending the file you should switch to tera term for it is a lot nicer on the eyes ;)

Just adding a warning on the github readme page would be enough and an upgrade guide on the openwrt wiki page.

I can help with that in the future ;) , I am sort of writing the steps down for myself right now anyway.

The forum topic about the 2812 is also to long and to dispersed to be easy useful for a new user. Again easily solved with a update to the installation page on openwrt. And this is not the place to discuss that ;)

**Comment by ScApi** [comment]
It was posted on forum that last revision was compiled without testing as my F1 died with surge. I will add info about this issue in readme.

Wiki was written when NAND/Device was in early support stage, now (except u-boot replacement part) You can flash it via initramfs and then via sysupgrade image but no one had time to do it + lede/openwrt transition and still wiki status is a bit unknown.

Teraterm takes about two minutes to upload asc in most cases, there is problem with some ttl adapters and/or drivers, I'm mostly using Linux so that part was suggested by oxo or Cornel.

Feel free to add/change anything on the wiki.

**Comment by Mafketel** [comment]
The wiki page has been updated ;)

Now we only got to tackle a bunch of other issues ;) this error still is current and there might be a resolution on its way https://github.com/openwrt/openwrt/pull/1883

p.s. not related but wanted to ask anyway, If by any change you remember how you sourced the rt3062.eeprom file that would be great. I do still have the impression it is hiding somewhere in the modem or on the original firmware from zyxel.
